src/main/java/com/powsybl/openloadflow/util/GeneratorState.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.util;
+
+import com.powsybl.iidm.network.Generator;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * @author Hadrien Godard <hadrien.godard at artelys.com>
+ */
+public class GeneratorState {
```

Comment:
You don't need that class, I think that all of this can be done inside `BusState`. with some adaptions in order not to save all the generators of a bus maybe.